analyzer/src/funTest/assets/projects/synthetic/carthage-expected-output.yml

```diff
@@ -0,0 +1,123 @@
+---
+project:
+  id: "Carthage:oss-review-toolkit:ort:"
+  definition_file_path: "<REPLACE_DEFINITION_FILE_PATH>"
+  declared_licenses: []
+  declared_licenses_processed: {}
+  vcs:
+    type: ""
+    url: ""
+    revision: ""
+    path: ""
+  vcs_processed:
+    type: "Git"
+    url: "<REPLACE_URL>"
+    revision: "<REPLACE_REVISION>"
+    path: "<REPLACE_PATH>"
+  homepage_url: ""
+  scopes: []
+packages:
+- id: "Carthage::binaryspec:1.0.0"
+  purl: "pkg:carthage/binaryspec@1.0.0"
+  declared_licenses: []
+  declared_licenses_processed: {}
+  description: ""
+  homepage_url: ""
+  binary_artifact:
+    url: "https://path/to/binary/artifact.zip"
+    hash:
+      value: ""
+      algorithm: ""
+  source_artifact:
+    url: "https://path/to/binary/artifact.zip"
```

Comment:
Why is the `binary` artifact path used?

Reply:
This is explained in [Carthage.kt:228](https://github.com/oss-review-toolkit/ort/pull/3099/files#diff-ff7d2fff941c8b35e59a6c2c4d1e17f7R228-R229)
```
        // The content from the artifactUrl might contain header files that should be scanned.
        // Therefore adding this URL for both sourceArtifact and binaryArtifact.
```
Do you disagree with that? What do you suggest?

As an example the carthage binary dependency for firebase analytics can be downloaded from here:
https://dl.google.com/dl/firebase/ios/carthage/6.9.0/Analytics-8e4f2b0b33540341.zip

Reply:
Thanks! Hadn't noticed above code comment. 
I believe until now the assumption hold that, when set, the source artifact and VCS info would point to a location containing a superset of the sources from which the artifact was build. In the case of the binary artifact you say that there are header files which should be scanned, thus these are in general not the whole sources.

With the implementation `as-is` when examining such dependency with ORT, it would go unnoticed that not all sources have been scanned: e.g. there would be no error indicating that problem. Until now we relied on that error being present in such case.

Given that I'd say the source artifact URL should not be set. But let's get other opinions to avoid some back and forth.

Reply:
I gave this some thought again, too. While I still do believe it makes sense to scan header files included in binary artifacts, we will basically have the same problem e.g. in binary Conan packages. So the more generic solution would probably be to continue to only add binary artifacts as binary artifacts in ORT, and not additionally as source artifacts, but add to ORT the (optional) feature to also scan (some files from) binary artifacts. This would at least make sense in the case where only a binary artifact is available, so we do scan at least the header files.

To summarize, I'd now vote to only add the binary artifact here, and implement the feature to scan header files from binary packages at another place in the scanner / downloader, so the logic applies to all package managers.